lms/djangoapps/instructor_task/tasks.py

```diff
@@ -140,6 +146,11 @@ def calculate_grades_csv(entry_id, xmodule_instance_args):
     """
     # Translators: This is a past-tense verb that is inserted into task progress messages as {action}.
     action_name = ugettext_noop('graded')
+    TASK_LOG.info(
+        u"Task: %s, InstructorTask ID: %s, Task type: %s, Preparing for task execution",
+        xmodule_instance_args.get('task_id'), entry_id, action_name
+    )
+
```

Comment:
We are not sure where exactly grade task hangs or goes in bad state so tried to log every step to help debugging.